src/hotspot/cpu/riscv/interp_masm_riscv_64.cpp

```diff
@@ -743,64 +743,53 @@ void InterpreterMacroAssembler::unlock_if_synchronized_method(TosState state,
   pop(state);
 }
 
-// Support function for remove_activation & Co.
-void InterpreterMacroAssembler::merge_frames(Register Rsender_sp, Register return_pc,
-                                             Register Rscratch1, Register Rscratch2) {
-  // Pop interpreter frame.
-  ld_PPC(Rscratch1, 0, R1_SP_PPC); // *SP
-  ld_PPC(Rsender_sp, _ijava_state(sender_sp), Rscratch1); // top_frame_sp
-  ld_PPC(Rscratch2, 0, Rscratch1); // **SP
-#ifdef ASSERT
-  {
-    Label Lok;
-    ld_PPC(R0, _ijava_state(ijava_reserved), Rscratch1);
-    cmpdi_PPC(CCR0, R0, 0x5afe);
-    beq_PPC(CCR0, Lok);
-    stop("frame corrupted (remove activation)", 0x5afe);
-    bind(Lok);
-  }
-#endif
-  if (return_pc!=noreg) {
-    ld_PPC(return_pc, _abi_PPC(lr), Rscratch1); // LR
-  }
-
-  // Merge top frames.
-  subf_PPC(Rscratch1, R1_SP_PPC, Rsender_sp); // top_frame_sp - SP
-  stdux_PPC(Rscratch2, R1_SP_PPC, Rscratch1); // atomically set *(SP = top_frame_sp) = **SP
-}
-
 void InterpreterMacroAssembler::narrow(Register result) {
   Register ret_type = R5_scratch1;
-  ld_PPC(R5_scratch1, in_bytes(Method::const_offset()), R27_method);
-  lbz_PPC(ret_type, in_bytes(ConstMethod::result_type_offset()), R5_scratch1);
+  Register scratch = R6_scratch2;
+  ld(R5_scratch1, R27_method, in_bytes(Method::const_offset()));
+  lbu(ret_type, R5_scratch1, in_bytes(ConstMethod::result_type_offset()));
 
-  Label notBool, notByte, notChar, done;
+  Label notBool, notByte, notChar, byteOne, shortOne, done;
 
   // common case first
-  cmpwi_PPC(CCR0, ret_type, T_INT);
-  beq_PPC(CCR0, done);
+  li(scratch, T_INT);
+  beq(scratch, ret_type, done);
 
-  cmpwi_PPC(CCR0, ret_type, T_BOOLEAN);
-  bne_PPC(CCR0, notBool);
-  andi_PPC(result, result, 0x1);
-  b_PPC(done);
+  li(scratch, T_BOOLEAN);
+  bne(scratch, ret_type, notBool);
+  andi(result, result, 0x1);
+  j(done);
 
   bind(notBool);
-  cmpwi_PPC(CCR0, ret_type, T_BYTE);
-  bne_PPC(CCR0, notByte);
-  extsb_PPC(result, result);
-  b_PPC(done);
+  li(scratch, T_BYTE);
+  bne(scratch, ret_type, notByte);
+  // sign-extend lower 8 bits
+  li(scratch, 0x80);
+  bge(result, scratch, byteOne);
+  andi(result, result, 0xff);
+  j(done);
+  bind(byteOne);
+  li(scratch, 0xffffff00);
+  orr(result, result, scratch);
+  j(done);
```

Comment:
cpu/x86/interp_masm_x86.cpp, 
```
  bind(notBool);
  cmpl(rcx, T_BYTE);
  jcc(Assembler::notEqual, notByte);
  LP64_ONLY(movsbl(result, result);)
  NOT_LP64(shll(result, 24);)      // truncate upper 24 bits
  NOT_LP64(sarl(result, 24);)      // and sign-extend byte
  jmp(done);
```
`shll` -- logical shift left, `srai` -- arithmetic shift right. 
Can we implement this?